### PR TITLE
chore: remove pyee

### PR DESCRIPTION
**Pull request by kumaraditya303**
Replace pyee runtime dependency with a simple asyncio event emitter with tests added.

### CODE REVIEW

**Comment by mxschmitt** [comment]
what is the reason for that change, did you run into any issues?

File: playwright/_impl/_frame.py [hunk 1]
```diff
@@ -244,12 +243,12 @@ async def query_selector(self, selector: str) -> Optional[ElementHandle]:
         )
 
     async def query_selector_all(self, selector: str) -> List[ElementHandle]:
-        return list(
-            map(
-                cast(ElementHandle, from_channel),
-                await self._channel.send("querySelectorAll", dict(selector=selector)),
+        return [
```
Comment (kumaraditya303) [inline playwright/_impl/_frame.py:246]: Changed to list comprehension as mypy was complaining

File: playwright/_impl/_js_handle.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ async def json_value(self) -> Any:
 def serialize_value(value: Any, handles: List[JSHandle], depth: int) -> Any:
     if isinstance(value, JSHandle):
         h = len(handles)
-        handles.append(value._channel)
+        handles.append(cast(JSHandle, value._channel))
```
Comment (kumaraditya303) [inline playwright/_impl/_js_handle.py:96]: Mypy error

File: playwright/_impl/_network.py [hunk 3]
```diff
@@ -326,7 +326,9 @@ def expect_event(
             wait_helper.reject_on_event(
                 self, WebSocket.Events.Error, Error("Socket error")
             )
-        wait_helper.reject_on_event(self._parent, "close", Error("Page closed"))
+        wait_helper.reject_on_event(
```
Comment (kumaraditya303) [inline playwright/_impl/_network.py:329]: Mypy error

**Pull request closed without merging**